code/mission/import/xwinglib.cpp

```diff
@@ -582,5 +599,161 @@ bool XWingMission::load(XWingMission *m, const char *data)
+
+		switch (oj->object_type) {
+			case 0:
+				noj->objectType = XWMObjectType::oj_Mine1;
+				break;
+			case 1:
+				noj->objectType = XWMObjectType::oj_Mine2;
+				break;
+			case 2:
+				noj->objectType = XWMObjectType::oj_Mine3;
+				break;
+			case 3:
+				noj->objectType = XWMObjectType::oj_Mine4;
+				break;
+			case 4:
+				noj->objectType = XWMObjectType::oj_Satellite;
+				break;
+			case 5:
+				noj->objectType = XWMObjectType::oj_Nav_Buoy;
+				break;
+			case 6:
+				noj->objectType = XWMObjectType::oj_Probe;
+				break;
+			case 7:
+				noj->objectType = XWMObjectType::oj_Platform;
+				break;
+			case 8:
+				noj->objectType = XWMObjectType::oj_Asteroid1;
+				break;
+			case 9:
+				noj->objectType = XWMObjectType::oj_Asteroid2;
+				break;
+			case 10:
+				noj->objectType = XWMObjectType::oj_Asteroid3;
+				break;
+			case 11:
+				noj->objectType = XWMObjectType::oj_Asteroid4;
+				break;
+			case 12:
+				noj->objectType = XWMObjectType::oj_Asteroid5;
+				break;
+			case 13:
+				noj->objectType = XWMObjectType::oj_Asteroid6;
+				break;
+			case 14:
+				noj->objectType = XWMObjectType::oj_Asteroid7;
+				break;
+			case 15:
+				noj->objectType = XWMObjectType::oj_Asteroid8;
+				break;
+			case 16:
+				noj->objectType = XWMObjectType::oj_Rock_World;
+				break;
+			case 17:
+				noj->objectType = XWMObjectType::oj_Gray_Ring_World;
+				break;
+			case 18:
+				noj->objectType = XWMObjectType::oj_Gray_World;
+				break;
+			case 19:
+				noj->objectType = XWMObjectType::oj_Brown_World;
+				break;
+			case 20:
+				noj->objectType = XWMObjectType::oj_Gray_World2;
+				break;
+			case 21:
+				noj->objectType = XWMObjectType::oj_Planet_and_Moon;
+				break;
+			case 22:
+				noj->objectType = XWMObjectType::oj_Gray_Crescent;
+				break;
+			case 23:
+				noj->objectType = XWMObjectType::oj_Orange_Crescent1;
+				break;
+			case 24:
+				noj->objectType = XWMObjectType::oj_Orange_Crescent2;
+				break;
+			case 25:
+				noj->objectType = XWMObjectType::oj_Orange_Crescent3;
+				break;
+			case 26:
+				noj->objectType = XWMObjectType::oj_Orange_Crescent4;
+				break;
+			case 27:
+				noj->objectType = XWMObjectType::oj_Orange_Crescent5;
+				break;
+			case 28:
+				noj->objectType = XWMObjectType::oj_Orange_Crescent6;
+				break;
+			case 29:
+				noj->objectType = XWMObjectType::oj_Orange_Crescent7;
+				break;
+			case 30:
+				noj->objectType = XWMObjectType::oj_Orange_Crescent8;
+				break;
+			case 31:
+				noj->objectType = XWMObjectType::oj_Death_Star;
+				break;
+			default:
+				return false;
+		}
+
+		switch (oj->object_iff) { // seemingly not used
+			case 0:
+				noj->objectIFF = XWMCraftIFF::iff_default;
+				break;
+			case 1:
+				noj->objectIFF = XWMCraftIFF::iff_rebel;
+				break;
+			case 2:
+				noj->objectIFF = XWMCraftIFF::iff_imperial;
+				break;
+			case 3:
+				noj->objectIFF = XWMCraftIFF::iff_neutral;
+				break;
+		}
+
+		switch (oj->object_formation) {
```

Comment:
See the other review comments about object formation.  Note the special handling of bits 3 and 4: these are handled *independently*, so they need special bitwise checks.